<a href="https://colab.research.google.com/github/VitorFRodrigues/Polo_Tech_Americanas/blob/main/Polo_Tech/Modulo_04/zProva_Estatistica.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

__1.__ Identifique nas alternativas abaixo, a opção que contenha apenas métricas de posição:

- [ ] média, desvio padrão
- [ ] desvio padrão, variância
- [x] mediana, moda
- [ ] variância, mediana
- [ ] média, correlação de Pearson

## 

O principal objetivo da estatística descritiva é gerar **medidas que resumam** o conjunto de dados que estamos analisando, ou seja, medidas que descrevam **a distribuição** dos dados, de forma quantitativa. Temos dois tipos de métricas disponíveis: as métricas de **posição** e **dispersão**. Vamos conhecer melhor essas métricas a seguir:

**Métricas de Posição**: Estas são medidas que resumem algumas propriedades do conjunto de dados:

- Média
- Mediana
- Moda
- Quartis


---

__2.__ Dado o boxplot conforme a figura abaixo:

![BoxPlot](https://github.com/VitorFRodrigues/Polo_Tech_Americanas/blob/main/Polo_Tech/Modulo_04/imgs/BoxPlot.PNG?raw=1)

Como é definida a fórmula para os limites superior e inferior do boxplot que separa os dados da distribuição dos prováveis outliers?

- [ ] Q1 + 1.5(Q3 - Q1), Q3 - 1.5(Q3 - Q1)
- [ ] Q3 + 1(Q3 - Q1), Q1 - 1(Q3 - Q1)
- [ ] Q3 - 1.5(Q3 - Q1), Q1 + 1.5(Q3 - Q1)
- [x] Q3 + 1.5(Q3 - Q1), Q1 - 1.5(Q3 - Q1)
- [ ] 1.5Q3 + (Q3 - Q1), 1.5Q1 - (Q3 - Q1)


## 

![IQR](https://github.com/VitorFRodrigues/Polo_Tech_Americanas/blob/main/Polo_Tech/Modulo_04/imgs/IQR.png?raw=1)

---

__3.__ Rafael queria levar dinheiro na escola para comprar balas, mas sua mãe só deixou que ele levasse duas moedas para a escola. Sabendo que dentro do cofre do Rafael, temos as seguintes moedas:

- 10 centavos = 5 moedas
- 25 centavos = 3 moedas
- 50 centavos = 10 moedas
- 01 real = 2 moedas

Qual a probabilidade de Rafael retirar respectivamente uma moeda de 1 real e uma moeda de 25 centavos?

- [ ] 30,0 %
- [ ] 10,0 %
- [x] 1,6 %
- [ ] 15,8 %
- [ ] 5,3 %

In [4]:
# Probabilidade
P_1Real = 2/(5+3+10+2)

# Retirar 25 Cent depois que foi removido 1 Real 
P_25cent = 3/(5+3+10+1)

P_tot = P_1Real * P_25cent
print(f'Probabilidade para retirar 25 cent dado que foi retirado 1 real:\nP(.25|1) = {round(P_tot*100,1)}%')

Probabilidade para retirar 25 cent dado que foi retirado 1 real:
P(.25|1) = 1.6%


Pelo método convencional:
```python
# Espaço Amostral
S = {.10, .10, .10, .10, .10, .25, .25, .25, .50, .50, .50, .50, .50, .50, .50, .50, .50, .50, 1, 1}

# Probabilidade de retirar 1 real
P(1) = 2/(5+3+10+1) = 2/20 = 1/10

# Aqui calculamos a partir da probabilidade condicional
P(A ∩ B) = P(B | A)P(A) = P(0.25 | 1)P(1)

P(0.25 | 1) = 3/(5+3+10+1) = 3/19

P(A ∩ B) = P(B | A)P(A) = 3/19 * 1/10 = 3/190 = 1,6%
```
---

__4.__ Em uma filial de um banco, 40% dos clientes que vão ao caixa eletrônico sacam dinheiro do caixa. Utilizando de uma distribuição binomial, qual é a probabilidade de no máximo 12 clientes sacarem dinheiro em um dia que tivemos 30 clientes no caixa eletrônico?

- [x] 57,8 %
- [ ] 52,2 %
- [ ] 42,2 %
- [ ] 47,8 %
- [ ] 40,0 %

In [10]:
from math import factorial
import numpy as np

def binomial(n, p, k):
    C = (factorial(n) / (factorial(n - k) * factorial(k)))
    return C * np.power(p, k) * np.power(1 - p, n - k)

In [17]:
# número de clientes:
n = 30

# probabilidade de saque no caixa
p = 40/100

# número máximo de saque em dinheiro
k_max = 12

prob = 0
for k in range(0,k_max+1):
  prob += binomial(n, p, k)

# print do resultado teorico
print(f"P(0<=k<=12): {np.round(prob*100, 1)}%")

P(0<=k<=12): 57.8%


__5.__ Em um centro de distribuição logístico da Amazon, chegam cerca de 1.440 produtos por dia. Sabendo que segue uma distribuição de Poisson, qual a probabilidade de que em um minuto cheguem 3 produtos?

- [ ] 36,8 %
- [ ] 55,2 %
- [ ] 24,5 %
- [x] 6,1 %
- [ ] 18,4 %

In [16]:
def Poisson(lbd, k):
    return np.exp(-lbd) * (lbd**k) / factorial(k)

lbd = 1440/(24*60) # 1440 produtos/dia * 1 dia/24hr * 1hr/60min = 1440/(24*60) produtos/min

# frequência procurada
k = 3 # produtos/min

# print do resultado:
print(f"P(k = 3) = {round(Poisson(lbd,k)*100, 1)}%")

P(k = 3) = 6.1%


__6.__ Suponha que 60% das mulheres que compram kits de gravidez instantâneos estejam, de fato, grávidas. Para um kit de uma marca específica, se a mulher estiver grávida, o teste fornecerá resultado positivo 96% das vezes e negativo 4% das vezes (um “falso negativo"). Se ela não estiver grávida, o teste resultará positivo em 1% das vezes (um “falso positivo”) e negativo 99% das vezes . Suponha que um teste resulte positivo. Qual a probabilidade de que a mulher esteja realmente grávida?

- [ ] 77,7 %
- [ ] 63,2 %
- [ ] 39,4 %
- [ ] 86,4 %
- [x] 99,3 %

__Resolução:__

> Consideremos que 60% das pessoas estejam de fato grávidas: $P(C) = 0.6$

> Logo, 40% das pessoas não estão: $P(¬C) = 0.4$

Os Falsos positivos são 1%, ou seja 1% dos testes positivos na verdade não estão grávidas. Já o caso dos falsos negativos são 4%, ou seja 4% dos testes negativos na verdade a pessoa está grávida. Vamos ver esse resumo na tabela abaixo:


|                | Mulher Grávida | Mulher Não Grávida |
|----------------|----------------|--------------------|
| Teste positivo | 96 %           | 1 %                |
| Teste Negativo |  4 %           | 99 %               |


Fazendo os cálculos de acordo com a suposição:


|                    | Mulher Grávida                              | Mulher Não Grávida                               |
|--------------------|---------------------------------------------|--------------------------------------------------|
| **Teste positivo** | Prob. Gráv. e teste pos: 60% × 96% = 57,6%  | Prob Não Gráv. e teste pos: 40% × 1% = 0,4%      |
| **Teste Negativo** | Prob Gráv. e teste neg: 60% × 4% = 2,4%     | Prob Não Gráv. e teste neg: 40% × 99% = 39,6%    |


O evento desejado é um Verdadeiro positivo, ou seja, **teste positivo e a mulher grávida**.

O **total de possibilidades com relação ao teste positivo** seria a **soma da probabilidade de “grávida e teste pos”** (verdadeiro positivo) com **“não grávida e teste pos” (falso positivo)**, que resultará em 57,6% + 0,4%, igual a 58%.

Portanto, a chance de estar grávida se temos um teste positivo é de **57,6%/58% = 99,3%**. 

---

__7.__ Uma máquina de sorvete está regulada de modo a servir uma média de 80g por casquinha. Se a quantidade servida por casquinha seguir uma distribuição normal com desvio padrão de 8g, qual a porcentagem de casquinhas que conterão mais de 90g de sorvete?

- [ ] 26,6 %
- [ ] 73,4 %
- [ ] 10,6 %
- [ ] 50,0 %
- [ ] 89,4 %

__8.__ O tempo para fabricar um perfume em uma empresa é descrito por uma variável aleatória X, medida em horas, com distribuição normal de média mu = 7 e desvio padrão sigma = 2. Qual a probabilidade de que um perfume será finalizado entre 4 e 11 horas?

- [ ] 91,0 %
- [ ] 93,4 %
- [ ] 9,0 %
- [ ] 97,7 %
- [ ] 2,3 %

__9.__ Dados os tipos de amostragem e as definições dos mesmos, assinale a alternativa que indique cada tipo de amostragem com a respectiva definição:

Tipos de Amostragem:

1 - Amostragem Aleatória Simples

2 - Amostragem Estratificada

3 - Amostragem por Conglomerado

Definições:

I - Método de amostragem onde a população é dividida em grupos (estratos) segundo algumas características conhecidas sobre a população em estudo;

II - Método de amostragem que divide em subpopulações (conglomerados), onde escolhem aleatóriamente por amostragem simples as subpopulações e todos os elementos de uma subpopulação irão fazer parte da amostra;

III - Método de amostragem mais simples onde serão escolhidos aleatoriamente elementos com ou sem reposição.

Alternativas
- [ ] 1-II, 2-I, 3-III
- [ ] 1-II, 2-III, 3-I
- [ ] 1-III, 2-I, 3-II
- [ ] 1-I, 2-II, 3-III
- [ ] 1-III, 2-II, 3-I

__10.__ Um fabricante de refrigerante anunciou que o índice de cafeína do seu refrigerante de cola não é maior do que 35mg por lata. Um laboratório realizou 20 análises do índice de cafeína obtendo os resultados (37, 34, 36, 37, 35, 32, 30, 35, 34, 36, 39, 38, 34, 35, 31, 30, 33, 35, 36, 31). Sabe-se que o índice de cafeína do refrigerante de cola dessa marca se distribui normalmente com variância igual a 4. Qual é, respectivamente, a conclusão do teste e o valor crítico para este teste com nível de significância de 5%? Utilize a suposição de que queremos identificar se a média de cafeína, caso não seja 35mg, seja maior que esse valor (caso unicaudal do teste). Obs.: não utilize a função ztest ou t.

Alternativas

- [ ] Não Rejeita H0, X_critico = 34,40 mg
- [ ] Rejeita H0, X_critico = 34,40 mg
- [ ] Não Rejeita H0, X_critico = 35,74 mg
- [ ] Rejeita H0, X_critico = 35,74 mg

__11.__ Para que uma série temporal seja considerada estacionária ou parcialmente estacionária, qual das alternativas abaixo deve ser verdadeira?

Alternativas
- [ ] Média e Variância constante
- [ ] Nenhuma das anteriores
- [ ] Média constante
- [ ] Variância constante

__12.__ Escolha nas alternativas abaixo, qual é a componente de uma série temporal identificada como qualquer alteração irregular e inexplicável?

Alternativas
- [ ] Ciclos
- [ ] Sazonalidade
- [ ] Tendência
- [ ] Ruído Aleatório